src/cdk-experimental/scrolling/virtual-scroll-viewport.spec.ts

```diff
@@ -339,6 +325,132 @@ describe('CdkVirtualScrollViewport', () => {
                 `rendered content size should match expected value at offset ${offset}`);
       }
     }));
+
+    it('should work with an Observable', fakeAsync(() => {
+      const data = new Subject<number[]>();
+      testComponent.items = data as any;
+      finishInit(fixture);
+
+      expect(viewport.getRenderedRange())
+          .toEqual({start: 0, end: 0}, 'no items should be rendered');
+
+      data.next([1, 2, 3]);
+      fixture.detectChanges();
+
+      expect(viewport.getRenderedRange())
+          .toEqual({start: 0, end: 3}, 'newly emitted items should be rendered');
+    }));
+
+    it('should work with a DataSource', fakeAsync(() => {
+      const data = new Subject<number[]>();
+      testComponent.items = new ArrayDataSource(data) as any;
+      finishInit(fixture);
+      flush();
+
+      expect(viewport.getRenderedRange())
+          .toEqual({start: 0, end: 0}, 'no items should be rendered');
+
+      data.next([1, 2, 3]);
+      fixture.detectChanges();
+      flush();
+
+      expect(viewport.getRenderedRange())
+          .toEqual({start: 0, end: 3}, 'newly emitted items should be rendered');
+    }));
+
+    it('should trackBy value by default', fakeAsync(() => {
+      testComponent.items = [];
+      spyOn(testComponent.cdkForOfViewContainer, 'detach').and.callThrough();
+      finishInit(fixture);
+
+      testComponent.items = [0];
+      fixture.detectChanges();
+
+      expect(testComponent.cdkForOfViewContainer.detach).not.toHaveBeenCalled();
+
+      testComponent.items = [1];
+      fixture.detectChanges();
+
+      expect(testComponent.cdkForOfViewContainer.detach).toHaveBeenCalled();
+    }));
+
+    it('should trackBy index when specified', fakeAsync(() => {
+      testComponent.trackBy = i => i;
+      testComponent.items = [];
+      spyOn(testComponent.cdkForOfViewContainer, 'detach').and.callThrough();
+      finishInit(fixture);
+
+      testComponent.items = [0];
+      fixture.detectChanges();
+
+      expect(testComponent.cdkForOfViewContainer.detach).not.toHaveBeenCalled();
+
+      testComponent.items = [1];
+      fixture.detectChanges();
+
+      expect(testComponent.cdkForOfViewContainer.detach).not.toHaveBeenCalled();
+    }));
+
+    it('should recycle views when template cache is large enough to accommodate', fakeAsync(() => {
+      testComponent.trackBy = i => i;
+      spyOn(testComponent.cdkForOfViewContainer, 'createEmbeddedView').and.callThrough();
+      finishInit(fixture);
+
+      // Should create views for the initial rendered items.
+      expect(testComponent.cdkForOfViewContainer.createEmbeddedView).toHaveBeenCalledTimes(4);
+
+      (testComponent.cdkForOfViewContainer.createEmbeddedView as any).calls.reset();
```

Comment:
Rather than casting it to `any`, you can cast it to `jasmine.Spy`. Alternatively, when you do the `spyOn` above, Jasmine returns a reference to the spy which you can use instead of writing out `testComponent.cdkForOfViewContainer.createEmbeddedView`. Same goes for the other tests that follow the same pattern.